src/odemis/driver/test/technolution_test.py

```diff
@@ -1940,38 +1962,48 @@ def test_get_and_subscribe(self):
         possible to call get() and subscribe() at the same time."""
         field_images = (3, 4)
         self.counter = 0
-        self.counter2 = 0
 
         dataflow = self.mppc.data
         dataflow.subscribe(self.image_received)
 
         for x in range(field_images[0]):
             for y in range(field_images[1]):
+                # Reset the event that waits for the image being received (puts flag to False).
+                self._data_received.clear()
                 dataflow.next((x, y))
+                # Wait until single field image data has been received (image_received sets flag to True).
+                if not self._data_received.wait(self.timeout):
+                    self.fail("No data received after %d s for field %d, %d" % (self.timeout, x, y))
 
         # Acquire single field without unsubscribing listener (expect error)
         with self.assertRaises(Exception):
             image = dataflow.get()
 
-        time.sleep(1.5 * field_images[0] * field_images[1])  # Wait a bit to allow some processing and receive images.
         self.assertEqual(field_images[0] * field_images[1], self.counter)
         dataflow.unsubscribe(self.image_received)
 
+        counter_1 = self.counter
```

Comment:
I could not understand this assignment along with the 2 assert statements in line 2005
```
        self.assertEqual(field_images[0] * field_images[1], counter_1)
        self.assertEqual(field_images[0] * field_images[1], self.counter)
```

Reply:
I don't think it makes sense either, I'll remove it.